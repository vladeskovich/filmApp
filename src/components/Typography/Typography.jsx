import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Typography.scss';

const VARIANTS = {
  body: 'p',
  header1: 'h1',
};

const Typography = ({
  variant,
  className,
  children,
}) => {
  const classes = classNames(styles.typography, className);
  const Component = VARIANTS[variant];

  return (
    <Component
      className={classes}
    >
      {children}
    </Component>
  );
};

Typography.propTypes = {
  variant: PropTypes.oneOf(Object.keys(VARIANTS)),
  children: PropTypes.node,
  className: PropTypes.string,
};

Typography.defaultProps = {
  variant: 'body',
};

export default Typography;
