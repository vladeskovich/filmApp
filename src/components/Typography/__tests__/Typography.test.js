import React from 'react';
import Typography from '../Typography';

describe('Typography', () => {
  const props = {
    variant: 'header1',
    className: 'test-text',
    onClick: () => {},
  };

  it('should render Typography component without props', () => {
    const component = mount(<Typography>Ok</Typography>);
    expect(component).toMatchSnapshot();
  });

  it('should render Typography component with props', () => {
    const component = mount(<Typography {...props}>Ok</Typography>);
    expect(component).toMatchSnapshot();
  });

  it('should call onClick method', () => {
    const mockCallBack = jest.fn();

    const component = shallow(<Typography {...props} onClick={mockCallBack}>Ok</Typography>);

    expect(mockCallBack.mock.calls.length).toBe(0);
    component.find('h1').simulate('click');
    expect(mockCallBack.mock.calls.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
