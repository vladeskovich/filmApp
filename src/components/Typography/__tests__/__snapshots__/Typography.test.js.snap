// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should call onClick method 1`] = `
<h1
  className="Typography__typography_1E76fqYp test-text"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Ok
</h1>
`;

exports[`Typography should render Typography component with props 1`] = `
<Typography
  className="test-text"
  onClick={[Function]}
  variant="header1"
>
  <h1
    className="Typography__typography_1E76fqYp test-text"
    onClick={[Function]}
  >
    Ok
  </h1>
</Typography>
`;

exports[`Typography should render Typography component without props 1`] = `
<Typography
  variant="body"
>
  <p
    className="Typography__typography_1E76fqYp"
  >
    Ok
  </p>
</Typography>
`;
